### ISSUE TITLE
Specific layout/theme/style suggestions

### ISSUE THREAD

**Issue by ApolonTorq** [retypeapp/retype#75]
Reposted here as a feature or enhancement request.
### Discussed in https://github.com/retypeapp/retype/discussions/57

<div type='discussions-op-text'>

<sup>Originally posted by **ApolonTorq** July  1, 2021</sup>
[Discussion 5](https://github.com/retypeapp/retype/discussions/5) is relevant background to these suggestions. The suggestions are to have [project level config](https://github.com/retypeapp/retype/blob/main/configuration/project.md) options for the following:
- Font size.
- Line Height.
- Maximum widths for the left, middle and right panes. In particular, it would be preferable to be able to configure the middle pane to take up more of the browser width, if available, on larger screens.
- Minimum widths for the left and right panes - to tailor when they disappear because of lack of available width.
- Minimum content widths for the left and right panes - to tailor when horizontal scroll bars kick in because of deep hierarchies or long titles on content.

I may not have worded the suggestions in a way that matches the terms used in your implementation. They are not intended as the exact settings used. Its more an allowance for a general direction of suggested layout configuration. The general direction of these suggestions is to retain whatever the opinionated stance that Retype takes with layout/design but provide some configuration to control the **compactness** or **density** of the information content that is displayed. Some content is better suited to a higher density of text and some is not - depending on the intended audience and purpose. Note that the beta API functionality is an area that I think would benefit from much higher density of textual content.</div>

**Comment by fabriciomurta** [comment]
That'd be a lot of power to break Retype we give users but also the means to make the pages look _just_ the way you wanted it to. Thanks for the suggestion!

It will potentially be pretty tricky to manipulate the custom margins and keep the custom components we implemented looking right, so we'd need to tread here with much care!

**Comment by geoffreymcgill** [comment]
The new `theme` support available in Retype `v3.11.0` does not include the above configurability, but we will now be able to add relatively easily in a future release. If you're still interested in the functionality, please let us know and we will bump up the priority.

**Comment by gatodelcaribe** [comment]
@geoffreymcgill There's any way to test the new Theme support on 3.11? While Retype is awesome to just setup and forget about everything and just make documentation, for me, it's very important to bring some customization on themes, for example there are a lot of lovers of Catppucin themes.

Font size and custom typography is something that couple of friends are concern when I presented Retype.

**Comment by geoffreymcgill** [comment]
> There's any way to test the new Theme support on 3.11?

See https://github.com/retypeapp/retype/issues/42#issuecomment-2995265347 for testing the current `theme` functionality.

Mostly colors are only currently supported, but we should be able to easily add fonts, widths, spacing, alignment, etc in a future release. It just takes a lot of testing for each of those features, so for now we focused on getting color customization fully implemented. See comment above:

> The new theme support available in Retype v3.11.0 does not include the above configurability, but we will now be able to add relatively easily in a future release. If you're still interested in the functionality, please let us know and we will bump up the priority.

> for example there are a lot of lovers of Catppucin themes.

Thanks for pointing out Catppucin themes. I will check it out. Is there any particular Catppucin theme/palette implementation that you prefer? Any live example you can link me to?

**Comment by geoffreymcgill** [comment]
For the retype.com website, I did a quick apply of the `Latte` palette for the **Light** mode and `Frappé` palette for **Dark** mode by setting the following `base-color` and `base-bg` variables for each:

```yaml
theme:
  base:
    base-color: "#8839ef"
    base-bg: "#eff1f5"
  dark:
    base-color: "#ca9ee6"
    base-bg: "#303446"
```

![Image](https://github.com/user-attachments/assets/49ebce12-665f-4654-8e1a-f6bdcd420781)

![Image](https://github.com/user-attachments/assets/480c4f3b-4828-41e1-90ba-a66f6ac25ca7)

There are other adjustments that should be made to pick up the colors for the other variants, but you can a pretty decent implementation of the Catppuccin palette with setting just those two color variables.

I can also see a few areas within the theme implementation where we can improve, so I will prioritize those enhancements.

**Comment by geoffreymcgill** [comment]
I applied those colors to the [retype.com](https://retype.com/) website. Check it out and let me know what you think.

**Comment by gatodelcaribe** [comment]
Seems pretty good, I think there are some sides where it can be improvement to the contrast using the Catppuccin palette. 
When you pass the mouse to the menu is too light, the current selection / page seems fine. 
<img width="347" alt="Image" src="https://github.com/user-attachments/assets/91d6b306-0741-4eef-a025-a1e520a8e763" />

For dark mode the Frappé feels good. 

I just try it on my own server but i can't manage it changue de colors. I'm using this configuration: 
```
branding:
  title: Gato del Caribe
  label: Docs

theme:
  base:
    base-color: "#8839ef"
    base-bg: "#eff1f5"
  dark:
    base-color: "#ca9ee6"
    base-bg: "#303446"
```

**Comment by geoffreymcgill** [comment]
> I just try it on my own server but i can't manage it changue de colors. I'm using this configuration:

Your `theme` configuration looks correct, so it's likely either you do not have the "preview" release installed, or you are not running in "Pro" mode. 

Try the following:

### 1. Ensure "preview" release is stalled

If testing the new `theme` functionality on your local machine, ensure you are using the `@next` release. The `theme` functionality will be included in the `3.11.0` release, but the final release of `v3.11.0` has not been published. The final release of `v3.11.0` should be soon.

The current "preview" release is `v3.11.0-preview4` and can be installed using the following command:

```
npm install retypeapp@next --global
```

### 2. Start in "Pro" mode trial

Add the following setting to your project `retype.yml` file:

```yaml
start:
  pro: true
```

### 3. Retype Pro key

**If you have a Retype Pro key**, confirm it is installed on your local machine, see https://retype.com/guides/key-setup/.

You can confirm which version of Retype you are using by running the following command:

```
retype --version
```

For me, today, I'm running version `3.11.0-preview4` and the retype.com website is running the same, see:

https://github.com/retypeapp/retype/blob/3d4e9f14ad44e3d10141dba6078793aac4756d35/.github/workflows/retype-action.yml#L19

Hope this helps.

**Comment by geoffreymcgill** [comment]
> When I did try I got 3.11.0-preview4, on new global update got the preview5, and it's working.

Yes, we _just_ released a new preview with some updates. Version `v3.11.0-preview5` is the latest preview and likely will be the last preview before the final `v3.11.0` release.
